### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by ysc703** [rksltnl/Deep-Metric-Learning-CVPR16#18]
Hi
I am trying to use this method to my own datasets.
I have run the matlab code with Stanford_Online_Products dataset, and it seems work well.

And I have some questions as follow:
1. If the dataset is very large, such as 1 millon images, the sum of pos pairs and neg pairs will be more than 1 billion. Using the training batch size 128, it will take more than 70000 iterations to consume all samples. Is it true?
2. It said "the fully connected layer was initialized with random weights" in the paper, but I got the nan value of bottom_data1 when fineturning from my model. I changed it to initialize from the network pretrained， it work well and the test loss value decreases step by step. Is it right?

Thank you!

**Comment by rksltnl** [comment]
Hi, 
1. Depends on how fine grained your dataset is and how many positive pairs you can generate. If you look at the LMDB generation code, we use all possible positive pairs and equal number of negative pairs (subsampled to match number of positive pairs). If your dataset is fine grained, we won't be able to generate many positive pairs. For example, imagine you have 1M images, 2 images per class. Then you can only generate 500k positive pairs. In this case, you'll only sample 500k negative pairs.
2. Caffe will randomly initialize the weights for you with Xavier (at least in the current proto config) initialization. If you manually did your own random initialization, then all bets are off.
